### PR TITLE
Onboarding UI update

### PR DESCRIPTION
**Pull request by ComicSAS**
Onboarding UI update

### Description
- Added onboarding form page animation 
- Made other pages animations run only when page is visible for the user

### Checklist
- [x] I have ensured this PR does not contain any breaking changes for existing functionality;
- [x] I have added automated tests that cover the new behavior and removed obsolete tests and code;

### Tested On
- [ ] Emulator
- [x] Real device

### CODE REVIEW

File: presentation/src/test/java/com/shifthackz/aisdv1/presentation/screen/setup/ServerSetupViewModelTest.kt [hunk 1]
```diff
@@ -82,6 +84,12 @@ class ServerSetupViewModelTest : CoreViewModelTest<ServerSetupViewModel>() {
         } returns Single.just(mockHuggingFaceModels)
     }
 
+    @After
```
Comment (ShiftHackZ) [inline presentation/src/test/java/com/shifthackz/aisdv1/presentation/screen/setup/ServerSetupViewModelTest.kt:87]: What is the point of `unmockkAll()` here ?

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/screen/settings/SettingsScreen.kt [hunk 2]
```diff
@@ -577,48 +577,50 @@ private fun ContentSettingsState(
...
-        )
-        SettingsItem(
-            modifier = itemModifier,
-            loading = state.loading,
-            startIcon = Icons.Default.Code,
-            text = LocalizationR.string.settings_item_source.asUiText(),
-            onClick = { processIntent(SettingsIntent.LaunchUrl.OpenSourceCode) },
-        )
-        //endregion
+        if(!state.onBoardingDemo){
```
Comment (ShiftHackZ) [inline presentation/src/main/java/com/shifthackz/aisdv1/presentation/screen/settings/SettingsScreen.kt:580]: Why this check was removed ?

File: presentation/src/test/java/com/shifthackz/aisdv1/presentation/screen/setup/ServerSetupViewModelTest.kt (hunk 1 above)
Comment (ComicSAS) [reply presentation/src/test/java/com/shifthackz/aisdv1/presentation/screen/setup/ServerSetupViewModelTest.kt:87]: Added it with idea to possibly fix Unit-tests issue.

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/screen/settings/SettingsScreen.kt (hunk 2 above)
Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/screen/settings/SettingsScreen.kt:580]: It was not removed but added. Did this to hide `Info` region as it does not show to the user when onboarding. Updated onboarding UI to look like this:

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/screen/onboarding/page/FormPageContent.kt [hunk 3]
```diff
@@ -1,35 +1,48 @@
...
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
 import com.shifthackz.aisdv1.core.localization.R as LocalizationR
 
 @Composable
 fun FormPageContent(
     modifier: Modifier = Modifier,
-) = Column(
+    isPageVisible: Boolean = false,
+    ) = Column(
```
[the remark this answers is not in the document]
Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/screen/onboarding/page/FormPageContent.kt:40]: Fixed.

File: presentation/src/main/java/com/shifthackz/aisdv1/presentation/screen/onboarding/page/LocalDiffusionPageContent.kt [hunk 4]
```diff
@@ -32,7 +35,7 @@ import com.shifthackz.aisdv1.core.localization.R as LocalizationR
 @Composable
 fun LocalDiffusionPageContent(
     modifier: Modifier = Modifier,
-) = Column(
+    ) = Column(
```
[the remark this answers is not in the document]
Comment (ComicSAS) [reply presentation/src/main/java/com/shifthackz/aisdv1/presentation/screen/onboarding/page/LocalDiffusionPageContent.kt]: Fixed.

**Pull request merged**